### PR TITLE
chore: improve types (internal)

### LINKED ISSUE #43: Add Interfaces for each domain type used
## Summary

What:

Add an interface for each sub domain type used, so this can be used in it's respective method

Why:

So we can better type the return values from `sendWebSocketMessage`, it'll give us a clearer idea on what will be returned **for that respective method**

How:

```typescript
namespace Page {
  export interface createIsolatedWorld {
     executionContentId: number
  }
}

public evaluatePage(...) {
  ...
   const result = await this.sendWebSocketMessage(
        "Page.createIsolatedWorld",
        {
          frameId: this.frame_id,
        },
      ) as Page.createIsolatedWorld;
}
```

These must also account ERRORS (eg frameId is "blahblahblah")

### PR DESCRIPTION
**Pull request by ebebbington**
Fixes #43 

## Summary

* Didn't go with creating namespaces and interfaces for the domains (see issue), instead just make it more clear what the result is on our methods

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -24,6 +24,7 @@
 ---
```
Comment (crookse) [inline README.md:24]: did `deno fmt` format this file? lmao

File: src/headless_browser.ts [hunk 2]
```diff
@@ -228,12 +218,14 @@ export class HeadlessBrowser {
...
     const res = await this.sendWebSocketMessage("Runtime.evaluate", {
       expression: command,
-    });
-    this.checkForErrorResult((res as DOMOutput), command);
-    // Tried and tested, and `result` is `{result: { type: "boolean", value: false } }`
-    const exists = ((res as DOMOutput).result as SuccessResult)
-      .value as boolean;
-    if (exists !== true) {
+    }) as { // Tried and tested
+      result: {
```
Comment (crookse) [inline src/headless_browser.ts:222]: can we make this a `Result` type? not sure if you tried that already

File: README.md (hunk 1 above)
Comment (ebebbington) [reply README.md:24]: Yeah lol deno fmt now supports markdown files

File: src/headless_browser.ts (hunk 2 above)
Comment (ebebbington) [reply src/headless_browser.ts:222]: The thing is, the output from the events are all different and vary a lot which is why I went down the route of specifying the type per method - at first I did try make an interface for every event, but I decided against that and I’m trying to remember why

Comment (crookse) [reply src/headless_browser.ts:222]: just seeing if we could DRY up the code. no biggie

Comment (ebebbington) [reply src/headless_browser.ts:222]: Wait did you mean create a new interface called Result which this method will use (and any others that are casting to the same type)?

**Review by crookse — APPROVED** [review]

File: src/headless_browser.ts (hunk 2 above)
Comment (ebebbington) [reply src/headless_browser.ts:222]: Un-resolving because I think you make a good point

Comment (ebebbington) [reply src/headless_browser.ts:222]: Actually, going to re-resolve this as the output from all the messages can vary so much, that it's hard to say if any others will return the same amount of data, so `Result` could be misleading

**Pull request merged**